dagshub/auth/tokens.py

```diff
@@ -222,10 +231,23 @@ def is_valid_token(token: Union[str, Auth, DagshubTokenABC], host: str) -> bool:
             # 500's might be ok since they're server errors, so check only for 400's
             assert not (400 <= resp.status_code <= 499)
             if resp.status_code == 200:
```

Comment:
A 500 will not return the dictionary, so probably should change it to this now:
```suggestion
            assert resp.status_code == 200
```